docs/interfaces.md

```diff
@@ -12,184 +12,350 @@ Slots may support multiple connections to plugs.  For example the OS snap
 exposes the ``network`` slot and all applications that can talk over the
 network connect their plugs there.
 
-## Supported Interfaces - Basic
+The availability of an interface depends on whether snapd is running on a
+classic (eg, traditional desktop or server) or on a native system. Interfaces
+may also be implicit to the OS snap or implemented only via snaps providing the
```

Comment:
The term "implicit" is an irrelevant implementation detail that we shouldn't surface on the UI or the documentation. These are real plugs and slots on the core snap. We're creating them dynamically simply because it was easier for us to put the system in place this way, but we don't want to surface this fact to the user, and in fact it most probably will go away altogether in favor of having these interfaces defined and controlled in the core snap itself.